firmware/Core/Src/Controller.c

```diff
@@ -103,9 +104,49 @@ static Controller_State_t LogData_State(void)
     return CTRL_DO_MATH;
 }
 
+
+
+
 static Controller_State_t DoMath_State(void)
 {
-    return CTRL_ACTUATE_FRIDGE;
+    static float loopCounter;
+    static float comp_off_counter;
+    //get 5 temp readings
+    float temps[5] = {readings.t1, readings.t2, readings.t3, readings.t4, readings.t5}; //5 temp probe readings
+    float temp_diffs[5]; //to store deltas between probe readings and avg temp
+    float avgTemp = 0; //mean temperature
+    float fan_treshold = 1; //desired fan treshold (set to 1C for now, can be changed)
+
+    for (int i=0, i<5; i++)
+        avgTemp += temps[i] / 5; //calculate mean temp
+
+    //fan control
+    if(loopCounter % 100 ==0) //every 2 seconds
+        for (int i=0, i<5; i++)
+        {
+            temp_diffs[i] = temps[i] - avgTemp; //calculate deviation from mean temp.
+            if (temp_diffs[i] >= fan_treshold || temp_diffs[i] <= -1*fan_treshold)
+                ActuatorCommands.fan = FAN_ON; //if any probe exceeds treshold, turn on internal fan
```

Comment:
if you write it like this, you'll find that if the first probe exceeds threshold, you'll set the setting to ON but then if the next probe exceeds you'll set it back to OFF. Maybe you want to do something like break out of the loop as soon as you find 1 probe that exceeds threshold ?

Reply:
added break if treshold is exceeded